### DISCUSSION TITLE
Backwards conversion: MCAP to rosbag1

### DISCUSSION THREAD

**Discussion by Hommus** [foxglove/mcap#1583] [General]
I would like to use the MCAP CLI to convert data from .mcap to .bag (rosbag1) for the purpose of data verification. Thank you!

**Comment by amacneil** [comment]
Seems like a nice idea, assuming that the mcap file contains only ROS 1 channels.

**Comment by mica-angeli** [comment]
Just wanted to comment that my company could use this feature as well.  If we can convert MCAP files back to Bag files, we could potentially use it as our sole bag recording format

**Comment by jtbandes** [comment]
> for the purpose of data verification

What kind of verification are you hoping to perform on the .bag file?

**Comment by wkalt** [comment]
assuming the objective is to be able to go bag -> mcap -> bag and get an identical hashing bag out, some complications to consider will be,
* compression implementations will not always emit the same bytes for the same inputs. I think identical hashes are only likely to come out on uncompressed bag/mcap.
* connection ID numbering would need to be preserved going from bag -> mcap. In theory this could be an issue since bags use a uint32 and mcap uses a uint16. In fact, this is technically a problem in conversion today. In practice most bags have under uint16 (65535) connections.
* the bag header record reserves some empty space for tooling to fill in. Usually this space is just empty. I'm not sure if tooling actually does fill it in. We would need to figure out some place to stuff this in the mcap file though.
